src/components/Application.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react'
+import { useQueryParameters } from './App'
+import { useHistory, useLocation, useParams } from 'react-router-dom'
+
+type TParams = { appId: string; sectionName?: string; page?: string }
```

Comment:
Nice! Just curios what is the different between using `?` and not using it?

Reply:
The `?` means its an optional field. I thought this was necessary since `sectionName` and `page` aren't always available from the URL (e.g. on the application home page). 

I see now that it doesn't actually matter in this case, since the values are only ever read from `useParams` and not explicitly defined in the code. But without the `?`, if you tried to put `const myParams = {appId: "1234"}` Typescript complains about missing fields.

Reply:
Interesting! Thanks.